client.go

```diff
@@ -96,7 +97,13 @@ func HijackHttpRequest(options HijackHttpOptions) error {
 	clientconn := httputil.NewClientConn(dial, nil)
 	defer clientconn.Close()
 
-	clientconn.Do(req)
+	res, err := clientconn.Do(req)
+	if res.StatusCode >= http.StatusBadRequest {
```

Comment:
This will not work. The initial connection will be upgraded and we receive an `101` when it is successful. Should we then call the error handler in all cases where the status code is not `101`?